magento/pwa-studio
# fix: move cart creation out of trigger

## Description

For some reason the cart trigger owns cart creation. When the component mounts it invokes a "getCartDetails" action, which in turn checks for existence of a cart id and if it doesn't find one, calls the "createCart" action, which does another check for cart id and if it also doesn't find one, invokes a mutation to create a cart.

This could be simpler. We can move cart creation to the cart context provider.

## Related Issue





Closes [PWA-776](https://jira.corp.magento.com/browse/PWA-776).

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Clear cache and refresh. A cart id should be fetched (via createCart) from the context provider instead of the trigger.
2. Log in. In the dev tools, modify your `signin_token` to invalidate it (remove a character).
3. Refresh the page. You should see errors logged in console from graphql about unauthorized access, but then you should see those queries be remade with a valid cart id and the user token should be removed.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
